src/FluentSchema.test.js

```diff
@@ -186,6 +189,36 @@ describe('FluentSchema', () => {
     })
   })
 
+  describe('if', () => {
+    it('simple', () => {
+      expect(
+        FluentSchema()
+          .prop('prop')
+          .prop('extraProp')
+          .if({ properties: { prop: { maxLength: 5 } } })
+          .then(
```

Comment:
We should not expose a `then`  method to not conflict with promises, as there is a concept of a "thenable".

Maybe `if` has the signature: `.ifThenElse(a, b, c)`

Reply:
very good point! But ifThenElse seems a bit weird if you use just `ifThen` option 🤔

Reply:
maybe we could have both `ifThen` and `IfThenElse` still not great to me